woodwork/type_sys/inference_functions.py

```diff
@@ -79,11 +77,13 @@ def timedelta_func(series):
 
 
 class InferWithRegex:
-    def __init__(self, get_regex: Callable[[], str]):
+    def __init__(self, get_regex):
         self.get_regex = get_regex
 
     def __call__(self, series: pd.Series) -> bool:
-        regex = self.get_regex()
+        regex_list = []
+        for regex in self.get_regex:
+            regex_list.append(ww.config.get_option(regex))
```

Comment:
I had thought about suggesting something like this for your previous PR.  However, the reason I chose not to do that is that it tightly couples the functioning of the `InferWithRegex` class to the config API.  This only limits the usefulness of `InferWithRegex` in the future.  It also makes the overall machinery of the class harder to understand on its own.  I actually think there might be fewer code changes required overall if we just add something like this:
```python
ip_address_func = InferWithRegex(
    lambda: (
        "(" + ww.config.get_option('ipv4_inference_regex') +
        "|" + ww.config.get_option('ipv6_inference_regex') + ")"
    )
)
```
Then we don't have to limit the capabilities of `InferWithRegex`.  And anyone trying to understand it in the future might have an easier time.  For example, they might only wish to understand the `url_func` inference function.  Then they wouldn't have to worry about how two config values are formatted into a regex for `ip_address_func`.